scipy/_lib/_util.py

```diff
@@ -707,7 +707,7 @@ def _nan_allsame(a, axis, keepdims=False):
     return ((a0 == a) | np.isnan(a)).all(axis=axis, keepdims=keepdims)
 
 
-def _contains_nan(a, nan_policy='propagate', policies=None, *, xp=None):
+def _contains_nan(a, nan_policy='propagate', policies=None, *, xp_ok=False, xp=None):
```

Comment:
can you briefly explain the intended semantics of `xp_ok`?

Reply:
Temporarily, while `_axis_nan_policy` does not handle non-NumPy arrays, other functions that call `_contains_nan` want it to raise an error if `nan_policy='omit'` and `xp` is not `np`. 

`_xp_mean` supports `nan_policy='omit'` natively, so setting this to `True` prevents the error from being raised. 

In this name, `xp` was intended to imply `xp` other than NumPy. Other possibilities include `xp_omit_ok` and `non_numpy_omit_ok`. Or we could take another perspective for naming the variable... maybe consider the name to indicate whether the calling function implements `nan_policy='omit'` itself or whether this function should raise when `xp` is not NumPy and `nan_policy='omit'`. Feel free to suggest a preferred name. 

Since this function is private , the need for the argument is temporary, and the argument will probably only ever be needed by a handful of functions, I'll probably just change it. Another possibility is to eliminate the argument and just try/except the error as needed.

Reply:
Maybe I should just `try`/`except`

Reply:
I think the argument sounds fine. `xp_omit_okay` is probably the clearest name IMO. In any case, it would be nice to add a short docstring to explain. But feel free to just explain to reviewers each time if you would rather not add a docstring.